Customizing/global/plugins/Services/Cron/CronHook/EffectivenessAnalysisReminder/classes/ilEffectivenessAnalysisReminderDB.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/* Copyright (c) 1998-2016 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Databeseabstraction for cron job
+ */
+class ilEffectivenessAnalysisReminderDB {
+	const TABLE_NAME = "eff_analysis_maillog";
+
+	/**
+	 * @var ilDB
+	 */
+	protected $db;
+
+	public function __construct(ilDB $db) {
+		$this->db = $db;
+	}
+
+	/**
+	 * install the plugin
+	 */
+	public function install() {
+		$this->createTable();
```

Comment:
Ich hab mir das so angewöhnt. Zum einen hat man dann alles was mit DB zu tun hat in einer Datei und zum anderen in die dbupdate.php lesbarer.

Denke das ist so ne art "freiwillige" aber gern gesehene Richtlinie.